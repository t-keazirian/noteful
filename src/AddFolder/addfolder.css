.new-folder-container {
  display: flex;
  flex-direction: column;
  border: 2px solid black;
  margin: 20px;
  padding: 5px 10px;
}

h2 {
  margin: 10px;
  text-align: center;
}

.field {
  display: flex;
  flex-direction: column;
  text-align: center;
  justify-content: center;
}

.folder-submit-button {
  background-color:#1C2541;
  border-radius:21px;
  cursor:pointer;
  color: #5BC0BE;
  font-family:Arial;
  font-size:15px;
  padding:10px;
  text-decoration:none;
  text-shadow:0px 1px 0px #283966;
}

.folder-submit-button:hover {
  background-color:#476e9e;
  cursor: pointer;
}

.folder-submit-button:active {
  position:relative;
  top:1px;
}

.folder-submit-button:disabled {
  background-color: gray;
}