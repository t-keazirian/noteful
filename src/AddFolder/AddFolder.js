import React from 'react';
import ApiContext from '../Context/ApiContext';
import config from '../config';
import './addfolder.css';
import ValidationError from '../ValidationError/ValidationError';

class AddFolder extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      folder: {
        value: '',
        touched: false,
      }
    }
  }

  static contextType = ApiContext;

  handleSubmit = (event) => {
    event.preventDefault();
    const { folder } = this.state;
    const folderObject = {
      "name": folder.value
    }

    fetch(`${config.API_ENDPOINT}/folders`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(folderObject),
      }).then((res) => res.json())
        .then(resJson => {
      this.context.addFolder(resJson);
      this.props.history.push('/');
    })
  }

  updateFolder(folder) {
    this.setState({
      folder: {
        value: folder,
        touched: true
      }
    })
  }

  validateFolder() {
    const folder = this.state.folder.value.trim();
    if (folder.length === 0) {
      return 'Your folder must have a name.';
    }
  }

  render() {
   const folderError = this.validateFolder();

    return (
      <div className='new-folder-container'>
        <h2>Create a new folder:</h2>
        <form className='new-folder-form' onSubmit={e => this.handleSubmit(e)}>
          <div className='field'>
            <label htmlFor='folder-name-input'>Folder Name:</label>
            <input type='text' id='folder-name-input' name='folder-name-input' onChange={e => this.updateFolder(e.target.value)}/>
          </div>

      {this.state.folder.touched && <ValidationError message={folderError} />}

          <div className='button-div'>
            <button 
            type='submit' className='folder-submit-button'
            disabled={
              this.validateFolder()
            }
            >Add Folder</button>
          </div>
        </form>
      </div>
    )
  }
}

export default AddFolder;