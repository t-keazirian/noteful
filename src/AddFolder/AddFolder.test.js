import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AddFolder from './AddFolder';

describe('Add Folder tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddFolder />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<AddFolder />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })
})