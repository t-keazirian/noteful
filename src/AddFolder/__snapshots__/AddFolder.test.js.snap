// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Folder tests renders the UI as expected 1`] = `
<div
  className="new-folder-container"
>
  <h2>
    Create a new folder:
  </h2>
  <form
    className="new-folder-form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="folder-name-input"
      >
        Folder Name:
      </label>
      <input
        aria-label="Name for new folder"
        aria-required="true"
        id="folder-name-input"
        name="folder-name-input"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="button-div"
    >
      <button
        className="folder-submit-button"
        disabled="Your folder must have a name."
        type="submit"
      >
        Add Folder
      </button>
    </div>
  </form>
</div>
`;
